File: C Sharp/Problem_01.cs
using System;

class Problem_01{

    /*
    If we list all the natural numbers below 10 that are 
    multiples of 3 or 5, we get 3, 5, 6 and 9. 
    The sum of these multiples is 23.

    Find the sum of all the multiples of 3 or 5 below 1000.
    */

    static int SumOfMultiples(int N, int[] divisors){
        int Sum = 0;
        for (int i = 1; i < N; ++i){
            foreach (int divisor in divisors){
                if (i % divisor == 0){
                    Sum += i;
                    break;
                }
            }
        }
        return Sum;
    }

    public static void Solve(){
        int N = 1000;
        int[] divisors = {3,5};
        var Sum = SumOfMultiples(N, divisors);
        Console.WriteLine($"The Sum of Multiples of {divisors} is {Sum}.");
    }

}

File: Python/Problem_01.py
"""
If we list all the natural numbers below 10 that are 
multiples of 3 or 5, we get 3, 5, 6 and 9. 
The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.
"""

if __name__ == '__main__':
    # Brute-Force addition of suitable numbers
    sum_of_multiples = sum(
        n for n in range(1000) 
        if n % 3 == 0 or n % 5 == 0)
    print(f'The brute-force result is {sum_of_multiples}.')

    # Direct Calculation via 'Little Gauss Method' 
    # and Principle of Inclusion-Exclusion
    a,b,c = 999//3, 999//5, 999//15
    sum_of_multiples = 3*a*(a+1)//2 + 5*b*(b+1)//2 - 15*c*(c+1)//2
    print(f'The calculated sum is {sum_of_multiples}.')
